git-secrets

```diff
@@ -135,25 +142,26 @@ scan_between() {
   local combined_patterns=$(load_combined_patterns)
   [ -z "${combined_patterns}" ] && return 0
 
+  local scan_from=$(echo "${scan_range}" | sed -nE "s/^(.*)\.\.(.*)$/\1/p")
   local scan_to=$(echo "${scan_range}" | sed -nE "s/^(.*)\.\.(.*)$/\2/p")
-  is_first_commit=$(check_if_first_commit "${scan_to}")
-  if [[ ${is_first_commit} == true ]]; then
-    scan_range="4b825dc642cb6eb9a060e54bf8d69288fbee4904..HEAD"
-    # First commit is bottom commit
-    local -r commits_cmd="git log ${scan_range} --pretty=%h --reverse"
-  else
-    local -r commits_cmd="git log ${scan_range} --pretty=%h --reverse --ancestry-path"
-  fi
+  local -r commits_cmd="git log ${scan_range} --pretty=%h \
+    --reverse --first-parent --no-merges"
 
   # No commits means nothing to scan (OK)
-  IFS=$'\n' read -r -d '' -a commits < <( ${commits_cmd} && printf '\0' )
-  [ -z "${commits}" ] && return 0
-  unset IFS
+  to_and_from_identical=$(check_if_commits_identical \
+    "${scan_to}" "${scan_from}")
 
-  local -r current_branch=$(git branch --show-current)
+  if [[ ${to_and_from_identical} == true ]]; then
+    commits="${scan_from}"
+  else
+    IFS=$'\n' read -r -d '' -a commits < <( ${commits_cmd} && printf '\0' )
```

Comment:
I'm getting the following error when testing `HEAD~1..HEAD`:

```
/usr/local/bin/git-secrets: line 157: git log HEAD~1..HEAD --pretty=%h     --reverse --first-parent --no-merges: command not found
```

Reply:
Thanks @Nessex, will have a look into it. 
Does that command work for you in your terminal?

```
git log HEAD~1..HEAD --pretty=%h --reverse --first-parent --no-merges
```

Reply:
@Nessex When you have a moment, mind trying again please? Hoping d18bf0f fixes it 🤔

Reply:
Sorry, I've tested it out and it's working now. Thanks for the fix!